libr/core/cmd_debug.c

```diff
@@ -1017,39 +1031,39 @@ static int cmd_debug_map(RCore *core, const char *input) {
 		eprintf ("No debug region found here\n");
 		return false;
 	case 'i': // "dmi"
-		{ // Move to a separate function
-			RCoreBinFilter filter;
-			const char *libname = NULL, *symname = NULL, *mode = "", *a0;
-			ut64 baddr = 0LL;
-			char *ptr;
-			int i;
-
-			if (input[1]=='*') {
-				ptr = strdup (r_str_trim_head ((char*)input + 2));
-				mode = "-r ";
-			} else {
-				ptr = strdup (r_str_trim_head ((char*)input + 1));
-			}
-			i = r_str_word_set0 (ptr);
-			switch (i) {
-			case 2: // get symname
-				symname = r_str_word_get0 (ptr, 1);
-			case 1: // get addr|libname
-				a0 = r_str_word_get0 (ptr, 0);
-				addr = r_num_math (core->num, a0);
-				if (!addr || addr == UT64_MAX) {
-					libname = r_str_word_get0 (ptr, 0);
+		switch (input[1]) {
+		case 0:
+		case '*':
+			{
+				const char *libname = NULL, *symname = NULL, *mode = "", *a0;
+				ut64 baddr = 0LL;
+				char *ptr;
+				int i;
+				if (input[1]=='*') {
+					ptr = strdup (r_str_trim_head ((char*)input + 2));
+					mode = "-r ";
+				} else {
+					ptr = strdup (r_str_trim_head ((char*)input + 1));
 				}
-				break;
-			}
-			r_debug_map_sync (core->dbg); // update process memory maps
-			r_list_foreach (core->dbg->maps, iter, map) {
-				if (core->bin &&
-						((addr != -1 && (addr >= map->addr && addr < map->addr_end)) ||
-						 (libname && (strstr (map->name, libname))))) {
+				i = r_str_word_set0 (ptr);
+				switch (i) {
+				case 2: // get symname
+					symname = r_str_word_get0 (ptr, 1);
```

Comment:
I'll get rid of this comment, since i think it's pretty clear from the variable name